### ISSUE TITLE
ProductDisplayPageResponse Crashes on Empty Results

### ISSUE THREAD

**Issue by agrosner** [bazaarvoice/bv-android-sdk#20]
```
08-26 08:46:03.276 12457-12605$$$$$ E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
                                                                        Process: $$$$$$, PID: 12457
                                                                        java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
                                                                            at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
                                                                            at java.util.ArrayList.get(ArrayList.java:308)
                                                                            at com.bazaarvoice.bvandroidsdk.ConversationsAnalyticsManager.sendSuccessfulConversationsDisplayResponse(ConversationsAnalyticsManager.java:299)
                                                                            at com.bazaarvoice.bvandroidsdk.LoadCallDisplay$DisplayDelegateCallback.onResponse(LoadCallDisplay.java:56)
                                                                            at okhttp3.RealCall$AsyncCall.execute(RealCall.java:133)
                                                                            at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
                                                                            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                                                                            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                                                                            at java.lang.Thread.run(Thread.java:818)

```

The request was something of the sort (obfuscated any client info):
https://stg.api.bazaarvoice.com/data/products.json?Include=Reviews&Filter=Id:eq:SKU_null&appVersion=5.0&bvAndroidSdkVersion=4.1.2&buildNumber=50000&appId=$$$$$&apiversion=5.4&passkey=$$$$$&Sort_Reviews=SubmissionTime:asc&Stats=Reviews&Limit_Reviews=10
Invalid product id returning 0 results.

**Comment by agrosner** [comment]
Looks like this snippet in `ConversataionsAnalyticsManager`:
```
 if (conversationResponse instanceof ProductDisplayPageResponse) {
            ProductDisplayPageResponse productDisplayPageResponse = (ProductDisplayPageResponse) conversationResponse;
            Product product = productDisplayPageResponse.getResults().get(0);
            sendProductPageView(MagpieBvProduct.RATINGS_AND_REVIEWS, product);
        }
```
Doesn't do bounds checking.

**Comment by caseykulm** [comment]
Fixed in 4.1.3 release
